src/lib/toolbar.ts

```diff
@@ -1042,6 +1050,29 @@ export const toolbarConfig: Record<ToolbarModeName, ToolbarMode> = {
         description: 'Exit the current sketch',
         links: [],
       },
+      {
+        id: 'line',
+        onClick: ({ modelingSend }) =>
+          modelingSend({
+            type: 'equip tool',
+            data: { tool: 'dimension' },
+          }),
+        icon: 'line',
+        status: 'available',
+        title: 'Line',
+        hotkey: 'L',
+        description: 'Start drawing straight lines',
+        links: [],
+        isActive: ({ sketchSolveState }) => {
+          if (!sketchSolveState) {
+            return false
+          }
+          return (
+            sketchSolveState.matches('using tool') &&
```

Comment:
Do you find the early return here more clear than optional chaining `sketchSolveState`?


```suggestion
          return (
            sketchSolveState?.matches('using tool') &&
```

Reply:
I went with 
```ts
isActive: ({ sketchSolveState }) =>
          !!(
            sketchSolveState?.matches('using tool') &&
            sketchSolveState?.context?.sketchSolveTool === 'dimension'
          ),
```

Optional chaining returns undefined, when isActive needs boolean, which the early return resolved, but instead I went with your option with a `!!(...)` wrapper.